### PR TITLE
fix: ipfs links

### PR DESCRIPTION
**Pull request by solidovic**
### Description

- Fix the opening in new tab (IPFS version)
- Fix for withdrawals/claim is active nav link (IPFS version)
- Fix scroll to top (IPFS version)
- Fix(reward page): clipboard button (IPFS version)

### Checklist:

- [x] Checked the changes locally.
- [ ] Created / updated analytics events.
- [ ] Created / updated the technical documentation (README.md / [docs](https://docs.lido.fi/) / etc.).
- [ ] Affects / requires changes in other services (Matomo / Sentry / CloudFlare / etc.).

### CODE REVIEW

**Review by Tarens2 — APPROVED** [review]

File: shared/components/header/components/navigation/navigation.tsx [hunk 1]
```diff
@@ -45,7 +45,10 @@ export const Navigation: FC = memo(() => {
       {routes.map(({ name, path, icon }) => {
         const isActive =
           pathnameWithoutQuery === path ||
-          (path.length > 1 && pathnameWithoutQuery.startsWith(path));
+          (path.length > 1 && pathnameWithoutQuery.startsWith(path)) ||
+          // fix for withdrawals/claim
+          (pathname.indexOf('withdrawals') > -1 &&
```
Comment (DiRaiks) [inline shared/components/header/components/navigation/navigation.tsx:50]: Let's not hardcode the page name (withdrawals) in the code that is common to all routing

Comment (solidovic) [reply shared/components/header/components/navigation/navigation.tsx]: Fixed

**Review by DiRaiks — APPROVED** [review]

File: shared/components/link-ipfs.tsx [hunk 2]
```diff
@@ -13,16 +19,28 @@ export const LinkIpfs = ({ onClick, query, ...props }: Props) => {
   const push = usePrefixedPush();
   const { href } = props;
 
+  // Actual for click (opening in same tab)
   const handleClick: MouseEventHandler<HTMLAnchorElement> = useCallback(
     (event) => {
       event.preventDefault();
       void push(href, query);
+
+      if (typeof window !== 'undefined') {
```
Comment (Jeday) [inline shared/components/link-ipfs.tsx:28]: How can click event happen on server?

File: shared/components/link-ipfs.tsx [hunk 3]
```diff
@@ -13,16 +19,28 @@ export const LinkIpfs = ({ onClick, query, ...props }: Props) => {
...
+  // Actual for opening in new tab
+  const basedHashHref = useMemo(
+    () => getBasedHashHref(href, query),
+    [href, query],
+  );
+
   // TODO:
   // eslint-disable-next-line jsx-a11y/anchor-has-content,jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions
-  return <a {...props} onClick={handleClick} />;
+  return <a {...props} href={basedHashHref} onClick={handleClick} />;
```
Comment (Jeday) [inline shared/components/link-ipfs.tsx:45]: Maybe we can pass children separately and solve `jsx-a11y/anchor-has-content`

File: shared/components/link-ipfs.tsx [hunk 4]
```diff
@@ -1,6 +1,12 @@
...
+  ReactNode,
+  MouseEventHandler,
+  useCallback,
+  useMemo,
+} from 'react';
 
 import { usePrefixedPush } from 'shared/hooks/use-prefixed-history';
+import { getBasedHashHref } from 'utils/get-based-hash-href';
 
 type Props = {
```
Comment (Jeday) [inline shared/components/link-ipfs.tsx:11]: Let's do union with `Omit<React.ComponentProps<'a'>,'conflicting fields'>`

File: shared/components/link-ipfs.tsx (hunk 2 above)
Comment (solidovic) [reply shared/components/link-ipfs.tsx:28]: why we need it for server?

Comment (Jeday) [reply shared/components/link-ipfs.tsx:28]: I mean, if it cannot happen why we need `window !==`?

Comment (solidovic) [reply shared/components/link-ipfs.tsx:28]: I didn't check, but on dev-ipfs we have HMR (potential there are problems)

File: shared/components/link-ipfs.tsx (hunk 4 above)
Comment (solidovic) [reply shared/components/link-ipfs.tsx:11]: done without Omit<>

**Review by Jeday — APPROVED** [review]

**Review by Jeday — APPROVED** [review]

**Pull request merged**